src/integrationTest/groovy/wooga/gradle/node/NodeReleasePluginPublishSpec.groovy

```diff
@@ -176,4 +182,42 @@ class NodeReleasePluginPublishSpec extends GithubIntegrationSpec {
         "candidate" | "0.1.0-rc.1"
         "final"     | "0.1.0"
     }
+
+    @Unroll
+    def 'builds and and create #task #version with github release #released'() {
+        given: "the future npm artifact"
+        packageJsonFile.exists()
+
+        and: "some content to publish"
+        def content = createFile("content.json")
+        content.text = "hello world"
+
+        and:
+        git.add(patterns: ['.'])
+        git.commit(message: 'add files')
+
+        when: "run the publish task"
+        def result = runTasks(task)
+
+        then:
+        result.success
+        result.wasExecuted("node_publish")
+        result.wasExecuted("npm_publish")
+
+        then:
+        result.wasSkipped("nodeGithubRelease") == !released
+        hasReleaseByName(version) == released
+        print testRepo.listReleases()
+
+        if (released) {
```

Comment:
This whole check makes no sense.
`hasReleaseByName(version)` will already check if a release with a certain name is available.
It makes only sense to fetch the release here to verify other properties of the release like `tagName`, `body` etc.